cleanlab/filter.py

```diff
@@ -398,6 +404,182 @@ class 0, 1, ..., K-1.
+    num_to_remove_per_class: int = None,
+    min_examples_per_class=1,
+    confident_joint: Optional[np.ndarray] = None,
+    n_jobs: int = None,
+    verbose: bool = False,
+) -> np.ndarray:
+    per_class_issues = _find_multilabel_issues_per_class(
+        labels,
+        pred_probs,
+        return_indices_ranked_by,
+        rank_by_kwargs,
+        filter_by,
+        frac_noise,
+        num_to_remove_per_class,
+        min_examples_per_class,
+        confident_joint,
+        n_jobs,
+        verbose,
+    )
+    if return_indices_ranked_by is None:
+        return per_class_issues.sum(axis=1) >= 1  # type: ignore
+    else:
+        label_issues_list, labels_list, pred_probs_list = per_class_issues
+        label_issues_idx = reduce(np.union1d, label_issues_list)
+        num_classes = get_num_classes(labels=labels, pred_probs=pred_probs)
+        label_quality_scores = np.zeros(len(labels))
+        for i in range(0, num_classes):
+            label_quality_scores += get_label_quality_scores(
+                labels=labels_list[i],
+                pred_probs=pred_probs_list[i],
+                method=return_indices_ranked_by,
+                **rank_by_kwargs,
+            )
+        label_quality_scores /= num_classes
+        label_quality_scores_issues = label_quality_scores[label_issues_idx]
+        return label_issues_idx[np.argsort(label_quality_scores_issues)]
+
+
+def _find_multilabel_issues_per_class(
+    labels: list,
+    pred_probs: np.ndarray,
+    return_indices_ranked_by: Optional[str] = None,
+    rank_by_kwargs={},
+    filter_by: str = "prune_by_noise_rate",
+    frac_noise: float = 1.0,
+    num_to_remove_per_class: int = None,
+    min_examples_per_class=1,
+    confident_joint: Optional[np.ndarray] = None,
+    n_jobs: int = None,
+    verbose: bool = False,
+) -> Union[np.ndarray, Tuple[List[np.ndarray], List[Any], List[np.ndarray]]]:
+    """
+    Parameters
+    ----------
+    labels : list
+      List of noisy labels for multi-label classification where each example can belong to multiple classes (e.g. ``labels = [[1,2],[1],[0],..]`` indicates the first example in dataset belongs to both class 1 and class 2.
+      For multi-label settings, your `labels` should instead satisfy: ``len(set(k for l in labels for k in l)) == pred_probs.shape[1])``.
+
+
+    pred_probs : np.ndarray
+      An array of shape ``(N, K)`` of model-predicted probabilities,
+      ``P(label=k|x)``. Each row of this matrix corresponds
+      to an example `x` and contains the model-predicted probabilities that
+      `x` belongs to each possible class, for each of the K classes. The
+      columns must be ordered such that these probabilities correspond to
+      class 0, 1, ..., K-1. They need not sum to 1.0
+
+
+    return_indices_ranked_by : {None, 'self_confidence', 'normalized_margin', 'confidence_weighted_entropy'}, default=None
+      Refer to documentation for this argument in filter.find_label_issues() for details.
```

Comment:
FYI it is ok here because this is a helper method not displayed in our official documentation, but this is not the way to reference another function properly in the docstrings. You can refer to Development.md guide for an example of how to properly refer to other functions in docstrings, which will be important for any user-facing function.

This one can be left as is though